src/main/java/ecse428/peaceOfMinde/controller/WorkerController.java

```diff
@@ -0,0 +1,91 @@
+import java.util.List;
+
+@CrossOrigin(origins = "*")
+@RestController
+@RequestMapping("")
+@RequiredArgsConstructor
+public class WorkerController {
+
+    private final WorkerService workerService; // TODO
+    private final BuyerService buyerService;
+
+    /**
+     * View a specific ServiceRequest
+     * @param serviceOfferingId - Service Offering ID to View
+     * @return
+     * @throws PersonException
+     */
+    @GetMapping(value = { "/person/worker/getServiceOffering", "/person/worker/getServiceOffering" })
+    public ResponseEntity<?> getServiceOffering(@RequestBody Integer serviceOfferingId) throws PersonException {
+        try {
+            ServiceOffering serviceOffering = workerService.getServiceOffering(serviceOfferingId);
+            return new ResponseEntity<>(LibraryUtil.convertToDto(serviceOffering), HttpStatus.OK);
+        } catch (ServiceOfferingException serviceOfferingException) {
+            return new ResponseEntity<>(serviceOfferingException.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * View all ServicOfferings from a Buyer
+     * @param buyerId
+     * @return
+     * @throws PersonException
+     */
+    @GetMapping(value = { "/person/worker/getServiceOffering", "/person/worker/getServiceOffering" })
+    public ResponseEntity<?> getAllServiceOfferings(@RequestBody Integer buyerId) throws PersonException {
+        try {
+            List<ServiceOffering> serviceOfferings = buyerService.getAllServiceOfferings(buyerId);
+            List<ServiceOfferingDto> serviceOfferingDtos = new ArrayList<>();
+
+            // Convert all the serviceOfferings to DTOs
+            for (int i =0; i<serviceOfferings.size(); i++) {
+                ServiceOffering currServiceOffering = serviceOfferings.get(i);
+                ServiceOfferingDto serviceOfferDto = LibraryUtil.convertToDto(currServiceOffering);
+                serviceOfferingDtos.add(serviceOfferDto);
+            }
+
+            return new ResponseEntity<>(serviceOfferingDtos, HttpStatus.OK);
+        } catch (ServiceOfferingException serviceOfferingException) {
+            return new ResponseEntity<>(serviceOfferingException.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    /**
+     * TODO: Implement
+     * Apply to a service offering
+     * from Worker to Buyer
+     * @param serviceOffering
+     * @throws
+     */
+
+    /**
+    PutMapping(value = { "/person/worker/apply", "/person/worker/apply" })
+    public ResponseEntity<?> applyToServiceOffering(@RequestBody ServiceOffering serviceOffering) throws PersonException {
+        try {
+            // Send message to Buyer
+            // Apply to the serviceOffering
+
+     
+            return new ResponseEntity<>(applicationStatus, HttpStatus.OK);
+        } catch (ServiceOfferingException serviceOfferingException) {
+            return new ResponseEntity<>(serviceOfferingException.getMessage(), HttpStatus.BAD_REQUEST);
+        }
+    }
+
+     **/
```

Comment:
Does this need to be filled in?

Reply:
Yeah this is for the "apply to initiate chat"

Just added it so I can do it too later